37 put admininstitutionsid spec

ValterAro opened the pull request:
Added PUT operation for the admin/institutions/{id} path in the administrative.yml OpenAPI 3.0.3 specification.
#37 issue

martinkearn commented on src/CentOps.Api/Extensions/ServiceCollectionExtensions.cs:
```diff
@@ -97,5 +97,25 @@ public static void AddCosmosDbServices(this IServiceCollection services, IConfig
+                    options.AddDefaultPolicy(builder =>
+                    builder
+                        .WithOrigins(
+                            corsConfig?.AllowedOrigins != null && corsConfig.AllowedOrigins.Any()
```
Would this be more readable if the value of this was worked out in a separate var which is passed in here? I'm finding it a bit hard to read in-line.

guypritchard replied:
Yep - will do.

decodingahmed commented on src/CentOps.Api/Extensions/ServiceCollectionExtensions.cs:
```diff
@@ -97,5 +97,25 @@ public static void AddCosmosDbServices(this IServiceCollection services, IConfig
+                    options.AddDefaultPolicy(builder =>
+                    builder
+                        .WithOrigins(
+                            corsConfig?.AllowedOrigins != null && corsConfig.AllowedOrigins.Any()
+                            ? corsConfig.AllowedOrigins.ToArray()
+                            : new string[] { "*" })
```
1. If `AllowedOrigins` is null/empty, I think we should fall back to the actual host instead of `*` otherwise we might start accepting requests from anywhere, right?  Sorry if I'm misunderstanding this.

2. Also, could you please move the `AllowedOrigin` logic outside of the AddCors block so it is a little easier to read.

guypritchard replied:
We spoke about this:

Summary - having some sort of a default if the configuration specified is useful hear as it means the service will start.  The dumb thing was allowing * domains in that instance.  We agreed to specify `localhost` as a fallback.  It will allow the service to start without config - but it will allow integration tests to pass, but a browser will complain bitterly, but in a way which Web Developers will understand if they call this API from the 'wrong' domain.

decodingahmed commented on src/CentOps.Api/appsettings.Development.json:
```diff
@@ -16,5 +16,8 @@
     "UseInMemoryStore": false
+  },
+  "Cors": {
+    "AllowedOrigins": [ "*" ]
```
I think this should be the host

guypritchard replied:
See above comment - we're falling back to `localhost` a valid but better default for an unconfigured state.

decodingahmed replied:
I think this may have been missed. Could you change the `*` to `locahost` please?

guypritchard replied:
This is really only development configuration - I'm only going to see this when running locally.  I've changed for consistency with other configuration - it certainly won't hurt at this stage.

decodingahmed commented on src/CentOps.Api/appsettings.json:
```diff
@@ -14,5 +14,8 @@
     "AdminApiKey": ""
+  },
+  "Cors": {
+    "AllowedOrigins":  ["*"]
```
I think this may have been missed. Could you change the `*` to `locahost` please?
```suggestion
    "AllowedOrigins":  ["localhost"]
```

ValterAro committed:
Merge pull request #90 from buerokratt/main

Updating from the main branch

ValterAro committed:
Update administrative.yml spec

Added PUT operation for the admin/institutions/{id} path in the administrative.yml OpenAPI 3.0.3 specification.
